### DISCUSSION TITLE
Authelia TOTP more options

### DISCUSSION THREAD

**Discussion by TimothyWright95** [authelia/authelia#5935] [Ideas]
Started to create a feature request but saw that it might be better to add it here first.

Description
It would be nice to have all the TOTP settings accessible from the app config, currently there are only period, skew, and issuer. Currently digits, secret size, and algorithm are missing. As these are more advanced features that may break 2FA devices (some do not support a different number of digits or a different algorithm) they should be locked behind a "Show Advanced Options" box.

Use Case
Wanting to be more secure by have 8 digits using sha512

Additional Info
Should be fairly easy to add since there are already chart settings changing authelia options in the same place the new options exist. Namely issuer/period/skew.

**Comment by james-d-elliott** [comment]
The values are all available: https://www.authelia.com/configuration/second-factor/time-based-one-time-password/

**Reply by TimothyWright95** [reply]
Yes, they are available to authelia. That's not the issue. They are not available during the settup of the authelia chart in truenas.

**Reply by james-d-elliott** [reply]
You should probably let them know, we have no control over that. See [our chart](https://github.com/authelia/chartrepo/blob/d0a42e24e127506fbc0d0cf849c79f5c611d5789/charts/authelia/values.yaml#L533-L566).

**Reply by TimothyWright95** [reply]
Oops looks like I posted on the wrong github page. Consequence of having too many tabs open I guess. My bad.

**Reply by james-d-elliott** [reply]
No big deal, at least you have a clear direction now! Have a good day/night.
